lib/trino-filesystem/src/main/java/io/trino/filesystem/memory/MemoryFileSystemCacheConfig.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.memory;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.units.DataSize;
+import io.airlift.units.Duration;
+import io.airlift.units.MaxDataSize;
+import jakarta.validation.constraints.NotNull;
+
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+import static java.util.concurrent.TimeUnit.HOURS;
+
+public class MemoryFileSystemCacheConfig
+{
+    // Runtime.getRuntime().maxMemory() is not 100% stable and may return slightly different value over JVM lifetime. We use
+    // constant so default configuration for cache size is stable.
+    @VisibleForTesting
+    static final DataSize DEFAULT_CACHE_SIZE = DataSize.succinctBytes(Math.min(
+            Math.floorDiv(Runtime.getRuntime().maxMemory(), 20L),
+            DataSize.of(200, MEGABYTE).toBytes()));
+
+    private Duration ttl = new Duration(1, HOURS);
+    private DataSize maxSize = DEFAULT_CACHE_SIZE;
+    private DataSize maxContentLength = DataSize.of(8, MEGABYTE);
```

Comment:
how did you come with this number? do we see 8MB manifest files in the wild?

Reply:
Unrelated question - do we consider caching these files on the fly? Using ZSTD for speed/ratio?

Reply:
8MB is the default used by iceberg library's manifest file caching, I just matched that. All the metadata files in benchmark are well below this limit, I don't know if multi-MB manifests are common in real tables.

Reply:
> do we consider caching these files on the fly? Using ZSTD for speed/ratio?

Haven't tried that so far, might be an interesting follow-up if we find that we don't have enough space in cache.
The on-disk filesystem cache is also available as a potential solution for lack of in-memory cache space.

Reply:
To add a data point for Delta, it can easily produce checkpoint files which are 100s of MBs in size for tables with frequent commits and compaction. The issue is that Delta includes remove actions in its checkpoints. This is an optimization which is only meaningful for the VACUUM operation, but which unfortunately can affect read and write performance.